### PR TITLE
19: Added orders and transactions store from airswap-web

### LINKED ISSUE #19: Setup orders store
Very similar to orders store in airswap-web.

Should be able to `takeOrder` and then eventually do a `swapAnySender`.

### PR DESCRIPTION
**Pull request by piersss**
# Description

Fixes #19 

I copied the orders and transactions stores from airswap web. I removed all the unnecessary code like:

- Wrapping / unwrapping ETH
- Requesting orders from makers and selecting best option

Things that are left that we actually need are:
- Approving tokens
- Taking orders
- Logging, tracking and updating transactions

I didn't write the code so don't pester me about it. 😛 

# Checklist:

- [x] I have read and understood [CONTRIBUTING.MD](https://github.com/airswap/airswap-marketplace/blob/main/CONTRIBUTING.md)
- [x] I have included the ticket number in my PR
- [x] I have performed a self-review of my own code
- [x] I have commented the parts where my code is hard to understand

### CODE REVIEW

File: src/redux/stores/balances/balancesActions.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { BigNumber } from 'ethers';
+
+import { AppDispatch, RootState } from '../../store';
+import { setBalances } from './balancesSlice';
+
+export const incrementBalance = (tokenAddress: string, amount: string, subtract = false) => (dispatch: AppDispatch, getState: () => RootState): void => {
```
Comment (Goddak) [inline src/redux/stores/balances/balancesActions.ts:6]: :vomiting_face: These line lengths :roll_eyes:

File: src/redux/stores/transactions/swapEventSubscriber.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            ? 'last-look'
+            : 'request-for-quote',
+      }),
+    );
+
+    const { transactions } = store.getState();
+
+    const matchingTransaction = transactions.all.find((tx) => {
+      if (tx.protocol === 'last-look') {
+        // Last look transactions don't have a nonce, but the
```
Comment (Goddak) [inline src/redux/stores/transactions/swapEventSubscriber.ts:41]: but the...... but the what? :rofl:

File: src/redux/stores/transactions/wrapEventSubscriber.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { Web3Provider } from '@ethersproject/providers';
```
Comment (Goddak) [inline src/redux/stores/transactions/wrapEventSubscriber.ts:1]: imports linter error

File: src/redux/stores/transactions/wrapEventSubscriber.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  dispatch(
+    incrementBalance(
+      transaction.order.senderToken,
+      transaction.order.senderAmount,
+    ),
+  );
+  dispatch(
+    incrementBalance(
+      transaction.order.signerToken,
+      transaction.order.senderAmount,
```
Comment (Goddak) [inline src/redux/stores/transactions/wrapEventSubscriber.ts:29]: signerAmount?

**Review by Goddak — CHANGES_REQUESTED** [review]

**Comment by Goddak** [comment]
I was going to approve with the first two comments `wrapEventSubscriber` is the only reason for the fail

File: src/redux/stores/transactions/swapEventSubscriber.ts (hunk 2 above)
Comment (piersss) [reply src/redux/stores/transactions/swapEventSubscriber.ts:41]: Ah, we won't have last-look transactions so I can delete this

File: src/redux/stores/transactions/wrapEventSubscriber.ts (hunk 3 above)
Comment (piersss) [reply src/redux/stores/transactions/wrapEventSubscriber.ts:1]: What's the error?

Comment (piersss) [reply src/redux/stores/transactions/wrapEventSubscriber.ts:1]: Ah it was delayed for me. Just a sort warning.

File: src/redux/stores/transactions/wrapEventSubscriber.ts (hunk 4 above)
Comment (piersss) [reply src/redux/stores/transactions/wrapEventSubscriber.ts:29]: Removed this too, we don't need to listen to wrap events.

File: src/redux/stores/balances/balancesActions.ts (hunk 1 above)
Comment (piersss) [reply src/redux/stores/balances/balancesActions.ts:6]: Agreed

**Review by Goddak — APPROVED** [review]

**Pull request merged**